network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/configuration/WgTunnel.kt

```diff
@@ -50,15 +52,15 @@ interface WgTunnel {
      *
      * @param keyPair is the private/public key [KeyPair] to be used in the wireguard [Config]
      */
-    suspend fun createWgConfig(keyPair: KeyPair? = null): Result<Config>
+    suspend fun createWgConfig(keyPair: KeyPair? = null, fromFailure: Boolean = false): Result<Config>
 
     /**
      * Creates a new wireguard [Config], returns it and stores it internally.
      * If one exists, updates it and returns it and also stores it internally.
      *
      * @param keyPair is the private/public key [KeyPair] to be used in the wireguard [Config]
      */
-    suspend fun createAndSetWgConfig(keyPair: KeyPair? = null): Result<Config>
+    suspend fun createAndSetWgConfig(keyPair: KeyPair? = null, fromFailure: Boolean = false): Result<Config>
```

Comment:
Tbf, I don't like this `fromFailure` boolean value, In general I don't like flag-arguments. There's plenty of literature around that, one post I like about the topic is [this one](https://martinfowler.com/bliki/FlagArgument.html).

I have other proposals, let's discuss 1on1